### PR TITLE
Cleaning up parsing in ParseAdd, ParseEdit, ParseDelete and ParseLendBorrow

### PR DESCRIPTION
**Pull request by itszhixuan**
using a new class ParseIndividualValue, which will be called upon when parsing each individual input is required. Reduces repeated code in the 4 parse classes.

Other items tackled in this PR included minor bugs that were identified when input command is not accompanied with any parameters, thus causing the application to crash. Another item was the conversion of date in ParseEdit to LocalDate

### CODE REVIEW

**Comment by TzeLoong** [comment]
LGTM

File: src/main/java/seedu/parser/MainInputParser.java [hunk 1]
```diff
@@ -46,12 +53,22 @@ public static Command parseInputs(String userInput) {
         case OtherExpenditureCommand.COMMAND_WORD:
         case TransportExpenditureCommand.COMMAND_WORD:
         case TuitionExpenditureCommand.COMMAND_WORD:
-            ParseAdd prepareAddExpenditure = new ParseAdd(splitValues[INDEX_USERSTRING]);
+            ParseAdd prepareAddExpenditure;
```
Comment (dsicol) [inline src/main/java/seedu/parser/MainInputParser.java:56]: Nice! good bug catch

File: src/main/java/seedu/parser/ParseIndividualValue.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package seedu.parser;
+
+import seedu.exceptions.EmptyStringException;
+import seedu.exceptions.ExceptionChecker;
+
+public class ParseIndividualValue {
```
Comment (dsicol) [inline src/main/java/seedu/parser/ParseIndividualValue.java:6]: Nice abstraction

**Review by dsicol — APPROVED** [review]
LGTM

**Pull request merged**